### PR TITLE
fix(Scripts/BlackwingLair): Implemented threat mechanic of Chromaggus…

### PR DESCRIPTION
**Pull request by UltraNix**
…'s Time Lapse.

Fixes azerothcore/azerothcore-wotlk#11060

### CODE REVIEW

**Comment by UltraNix** [comment]
@temperrr Please, test it.

**Comment by Nefertumm** [comment]
Hmm I don't know if that's the right bracket to put it in. I added a new bracket here https://github.com/azerothcore/mod-progression-system/pull/181 for blackwing lair

**Comment by UltraNix** [comment]
So it's not live yet, so for now it is in that bracket. When your PR is merged, then move it to proper bracket.

**Comment by Nefertumm** [comment]
Fair enough!

**Comment by temperrr** [comment]
Ugh why does blizzard change mechanics. Thanks for double checking.

**Comment by temperrr** [comment]
It was difficult to test this ,but I think it worked.

**Comment by Nefertumm** [comment]
Move it to BWL bracket when you can, please! :D

**Comment by UltraNix** [comment]
@Nefertumm Done.
@temperrr Updated with https://github.com/azerothcore/azerothcore-wotlk/pull/11161.

**Comment by temperrr** [comment]
Cheers!

File: src/Bracket_60_2_A/scripts/BlackwingLair/boss_chromaggus.cpp [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+                        events.ScheduleEvent(EVENT_BREATH_2, 60000);
+                        break;
+                    case EVENT_AFFLICTION:
+                    {
+                        uint32 afflictionSpellID = RAND(SPELL_BROODAF_BLUE, SPELL_BROODAF_BLACK, SPELL_BROODAF_RED, SPELL_BROODAF_BRONZE, SPELL_BROODAF_GREEN);
+                        std::vector<Player*> playerTargets;
+                        Map::PlayerList const& players = me->GetMap()->GetPlayers();
+                        for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+                        {
+                            if (!player->IsGameMaster() && !player->IsSpectator() && player->IsAlive())
```
Comment (Nefertumm) [inline src/Bracket_60_2_A/scripts/BlackwingLair/boss_chromaggus.cpp:274]: player not declared, build issue. Also, use ranged for as below for this :D

**Comment by Nefertumm** [comment]
Also update with https://github.com/azerothcore/azerothcore-wotlk/pull/11254 when you can, thank you :D

File: src/Bracket_60_2_A/scripts/BlackwingLair/boss_chromaggus.cpp (hunk 1 above)
Comment (UltraNix) [reply src/Bracket_60_2_A/scripts/BlackwingLair/boss_chromaggus.cpp:274]: Ranged for is not suitable for this.

**Comment by UltraNix** [comment]
@Nefertumm Done.

**Comment by Nefertumm** [comment]
Ty! Can you merge it when the build is done?

**Comment by UltraNix** [comment]
Okay.

**Pull request merged**